x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -645,6 +647,26 @@ export const LensTopNavMenu = ({
                 title: i18n.translate('xpack.lens.app.shareModal.title', {
                   defaultMessage: 'Share this Lens visualization',
                 }),
+                config: {
+                  link: {
+                    draftModeCallOut: (
+                      <EuiCallOut
+                        color="warning"
+                        title={
+                          <FormattedMessage
+                            id="xpack.lens.app.shareModal.draftModeCallout.title"
+                            defaultMessage="Unsaved changes"
+                          />
+                        }
+                      >
+                        <FormattedMessage
+                          id="xpack.lens.app.shareModal.draftModeCallout.link.warning"
+                          defaultMessage="The link will only work temporarily. Save the Lens visualization first to create a permanent link."
```

Comment:
Just one question: what do you mean by temporarily? What are the actions that can invalidate the link?

Reply:
I think this is wrong.
The visualisation is ephemeral, its state is encapsulated into the shortURL link itself and nowhere else. If the link is lost, so its the visualization.
But the link will always work.
The only exception is that the shortURL SO is deleted, then the link becomes invalid and an empty editor would be opened.

Reply:
this will happen on every shortURL SO I believe, if its deleted every link to that shortUrl can't be linked to something

Reply:
@eokoneyo as @dej611 said, the message here is not correct.
If you have unsaved changes, the visualization state will be always saved within the `shortUrl` SO. So until this SO will remain available the link will continue to work.
The Message here should probably revisited. If you have unsaved changes you are sharing the current snapshot of the visualization configurations and not a permanent link to the visualization.

Reply:
Hi @markov00, @dej611  thanks for the clarification, the update to the text came as a result of some tweaks we've been making to the messaging for the share modal. We previously had the text `Copy the link to get a temporary link. Save the lens visualization to create a permanent link.` prior to this particular text. 

I wonder especially given that the link the user will be copying is guaranteed to always work unless the backing  SO object is deleted, is there actually a need to display this message?

Reply:
Hi @markov00 @dej611 , I made an adjustment to the text we display to user for the lens context, that I think  communicates the behaviour of the link, it reads like this; 

<img width="544" alt="Screenshot 2024-12-10 at 13 46 04" src="https://github.com/user-attachments/assets/68e6160d-2048-429f-a366-05f66456bfa5">


Let me know what you think

Reply:
Sounds good to me, thanks for this rewording!